src/components/FunctionalComponents/NextService/NextService.js

```diff
@@ -0,0 +1,230 @@
+    e.stopPropagation()
+    const changedValue = e.target.value
+    if (changedValue) {
+      const newInterval = {
+        ...componentServices[serviceName],
+        lastService: changedValue,
+      }
+      const newComponentServices = {
+        ...componentServices,
+        [serviceName]: newInterval,
+      }
+      const newServiceIntervals = {
+        ...serviceIntervals,
+        [componentName]: newComponentServices,
+      }
+      setServiceIntervals(newServiceIntervals)
+    }
+    setLastServiceInputVisible(false)
+  }
+
+  function handleChangeInterval(e) {
+    e.stopPropagation()
+    e.preventDefault()
+    const changedValue = e.target.elements.changeIntervalInput.value ?? null
+    if (changedValue) {
+      const newInterval = {
+        ...componentServices[serviceName],
+        interval: changedValue,
+      }
+      const newComponentServices = {
+        ...componentServices,
+        [serviceName]: newInterval,
+      }
+      const newServiceIntervals = {
+        ...serviceIntervals,
+        [componentName]: newComponentServices,
+      }
+      setServiceIntervals(newServiceIntervals)
+    }
+    setIntervalInputVisible(false)
+  }
+
+  function calculateProgress() {
+    const ridesSinceService = rides.filter(ride => ride.date >= lastService)
+    const total =
+      unit === 'hrs'
+        ? sumDurations(ridesSinceService).hours
+        : Math.floor(sumDistances(ridesSinceService))
+    const width = (total / interval) * 100
+    return { width, total }
+  }
+}
+
+const ServiceCaption = styled.h3`
+  display: inline;
+  margin: 10px 0 0;
+  font-weight: normal;
+  width: 100%;
+  font-size: 125%;
+  text-align: center;
+`
+
+const LastServiceCaption = styled.figcaption`
+  color: var(--highlight);
+  margin: 10px 0;
+  font-weight: normal;
+  width: 100%;
+`
+
+const LastServiceDate = styled.figure`
+  position: relative;
+  margin: 0;
+  font-size: 112.5%;
+`
+
+const Date = styled.time`
+  text-align: start;
+  width: 100%;
+`
+
+const EditLastService = styled.input`
+  position: absolute;
+  padding: 5px;
+  left: 25px;
+  height: max-content;
+  bottom: 0;
+  width: min-content;
+`
+
+const NextServiceProgress = styled.figure`
+  position: relative;
+  margin: 0 0 20px;
+`
+
+const ProgressCaption = styled.figcaption`
+  text-align: end;
+`
+
+const EditInterval = styled.input`
+  &::-webkit-outer-spin-button,
+  &::-webkit-inner-spin-button {
+    -webkit-appearance: none;
+    margin: 0;
+  }
+
+  -moz-appearance: textfield;
+  position: absolute;
+  padding: 5px;
+  right: 24px;
+  height: max-content;
+  top: -5px;
+  text-align: start;
+  width: 100px;
+`
+
+const ConfirmButton = styled.button`
+  position: absolute;
+  top: -12px;
+  right: 50px;
+  background: transparent;
+  box-shadow: none;
+  color: var(--dark);
+`
+const CancelButton = styled.button`
+  position: absolute;
+  top: -12px;
+  right: 20px;
+  background: transparent;
+  box-shadow: none;
+  color: var(--dark);
+`
```

Comment:
These styles for the buttons are almost identical. You could think about having the styles once and then reuse the component.
```suggestion
const ConfirmButton = styled.button`
  position: absolute;
  top: -12px;
  right: 50px;
  background: transparent;
  box-shadow: none;
  color: var(--dark);
`
const CancelButton = styled(ConfirmButton)`
  right: 20px;
`
```